content/6_Advanced/BCC_2CC.mdx

```diff
@@ -25,7 +25,73 @@ frequency: 1
 
 <Problems problems="sam2" />
 
-(implementation)
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+const int NMAX = 3e5;
+
+int n, m, timer, SCC;
+int id[NMAX], low[NMAX];
+vector<int> g[NMAX], two_edge_components[NMAX];
+stack<int> st;
+
+void dfs(int node, int parent = -1) {
+	id[node] = low[node] = ++timer;
+	st.push(node);
+	bool pp = 0;
```

Comment:
what this variable do

Reply:
The graph may not be simple. So it can be multiple edges between 2 nodes.  I can change the answer.

Reply:
> it can be multiple edges between 2 nodes

what can be?

> it can change the answer?

what's "it"? how exactly can it change the answer

Reply:
The task is to split the graph into components by bridges/ edges of failure (if removed we get 2 components). If a node had has more than 1 edge with its father, then it's surely in the same component with its father. If it has only 1 edge it may be other component

Reply:
I named it  `multiple_edges`